File: src/test/java/de/moritzf/quota/idea/QuotaUiUtilTest.java
package de.moritzf.quota.idea;

import org.junit.jupiter.api.Test;

import java.time.Instant;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;

class QuotaUiUtilTest {
    @Test
    void formatResetCompactReturnsNullForNullReset() {
        assertNull(QuotaUiUtil.formatResetCompact(null));
    }

    @Test
    void formatResetCompactReturnsRelativeValueForFutureReset() {
        String formatted = QuotaUiUtil.formatResetCompact(Instant.now().plusSeconds(120));

        assertNotNull(formatted);
        assertFalse(formatted.startsWith("in "));
    }

    @Test
    void formatResetCompactReturnsNullForPastReset() {
        assertNull(QuotaUiUtil.formatResetCompact(Instant.now().minusSeconds(60)));
    }
}

File: src/test/kotlin/de/moritzf/quota/idea/QuotaUiUtilTest.kt
package de.moritzf.quota.idea

import de.moritzf.quota.idea.ui.QuotaUiUtil
import kotlin.time.Clock
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertNotNull
import kotlin.test.assertNull
import kotlin.test.assertTrue
import kotlin.time.Duration.Companion.seconds

class QuotaUiUtilTest {
    @Test
    fun formatResetCompactReturnsNullForNullReset() {
        assertNull(QuotaUiUtil.formatResetCompact(null))
    }

    @Test
    fun formatResetCompactReturnsRelativeValueForFutureReset() {
        val formatted = QuotaUiUtil.formatResetCompact(Clock.System.now().plus(120.seconds))

        assertNotNull(formatted)
        assertFalse(formatted.startsWith("in "))
    }

    @Test
    fun formatResetCompactReturnsNullForPastReset() {
        assertNull(QuotaUiUtil.formatResetCompact(Clock.System.now().minus(60.seconds)))
    }

    @Test
    fun formatExpiryUsesExpiresNotResets() {
        val formatted = QuotaUiUtil.formatExpiry(Clock.System.now().plus(120.seconds))
        assertNotNull(formatted)
        assertTrue(formatted.startsWith("Expires in "))
        assertFalse(formatted.startsWith("Resets "))
    }

    @Test
    fun formatOpenCodeBalanceConvertsFractionalUnitsToDollars() {
        assertEquals("12.35", QuotaUiUtil.formatOpenCodeBalance(1_234_567_890L))
        assertEquals("0.00", QuotaUiUtil.formatOpenCodeBalance(0L))
        assertEquals("0.01", QuotaUiUtil.formatOpenCodeBalance(1_000_000L))
        assertEquals("100.00", QuotaUiUtil.formatOpenCodeBalance(10_000_000_000L))
    }

    @Test
    fun escapeHtmlEscapesSpecialCharacters() {
        assertEquals(
            "&lt;script&gt;&amp;&quot;&#39;&lt;/script&gt;",
            QuotaUiUtil.escapeHtml("<script>&\"'</script>"),
        )
    }
}
